### ISSUE TITLE
Bugs in fill_outline shader

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/mapbox-gl-native#8751]
The fragment shader does `smoothstep(1.0, 0.0, dist)`. But [`smoothstep`'s behavior is undefined](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/smoothstep.xhtml) when its first argument is >= its second argument. On macOS, it appears that the result is always 0.

Changing it to `smoothstep(0.0, 1.0, dist)` also doesn't work -- except for a band across the middle of the screen, the result value is always 1.0.

Using the following shader modification, you can see it's calculating the distance from the a horizontal line through the center of the viewport, rather than the desired value:

```
    float alpha = dist / 100.0;
    gl_FragColor = vec4(alpha, alpha, alpha, 1.0);
```

![image](https://cloud.githubusercontent.com/assets/98601/25059527/fd533074-213c-11e7-94b3-930c22e8140f.png)

cc @kkaefer @ansis

**Comment by kkaefer** [comment]
Looks like this was broken from the very beginning: https://github.com/mapbox/mapbox-gl-js/commit/63e9f71df34958e16c8aa0e3d19da307da0360bf#diff-97267ac172ce8fd831ea9c2993add6a6R9

**Comment by kkaefer** [comment]
This is almost certainly the culprit for https://github.com/mapbox/mapbox-gl-native/issues/1555

**Comment by kkaefer** [comment]
> On macOS, it appears that the result is always 0.

How did you reach that conclusion? I'm seeing different results (no antialiasing) when manually setting `float alpha = 0.0`.

**Comment by jfirebaugh** [comment]
Fixed in #9699.
